pkg/utils/shared_objects/host_symbols_loader.go

```diff
@@ -0,0 +1,131 @@
+package shared_objects
+
+import (
+	"debug/elf"
+	"github.com/hashicorp/golang-lru/simplelru"
+)
+
+// HostSharedObjectSymbolsLoader is responsible for efficient reading of shared object's symbols.
+// The logic of the loader here is used on absolute paths, so container relative paths won't work here.
+type HostSharedObjectSymbolsLoader struct {
+	loadingFunc func(path string) (*soDynamicSymbols, error)
+	soCache     soDynamicSymbolsCache
+}
+
+func InitSODynamicSymbolsLoader(cacheSize int) *HostSharedObjectSymbolsLoader {
```

Comment:
I assume this may be used alongside the container symbols loader? So this should probably just be `HostSymbolsLoader` leave the interface name to the interface, IMO concrete implementations should present themselves as such.